### ISSUE TITLE
[Script Canvas] Quick way to go to a SC node source code

### ISSUE THREAD

**Issue by ppinfel** [o3de/o3de#3489]
### Summary:
I want a quick way to go to the source code C++ implementation of a Script Canvas node

### What is the motivation for this suggestion?
As a O3DE game developer that isn't proficient in C++, one of the best ways for me to learn C++ is to view the source code implementation of SC nodes that I am familiar with and learning by existing examples. 

Usually the most important or most used functionality will already be implemented as a SC node, but for Lua it is not clear how to do the same. With this feature you can quickly view what is going on behind an SC node to have some idea how to do it in Lua.

This is mostly a convenience feature but I think this will encourage more people to delve into C++ and motivate more contributions to SC from the open source community.

### Suggestion design description:
Right clicking on a Script Canvas node will additionally have the option of something like "open source file".

**Comment by monroegm** [comment]
We need to figure out how we would accomplish something like this @lsemp3d @carlitosan

**Comment by lsemp3d** [comment]
I don't believe we can automatically get that information for the vast majority of nodes because they come from the Behavior Context, we could definitely do it for the custom nodes, but they'll be a small subset of the entire library. We could leverage the text translation system to add custom hints, but they would need to be manually added and maintained.

**Comment by lsemp3d** [comment]
I recently added a tooltip addition that shows the C++ counterpart to any given node, while it doesn't automatically take you there, it gives you the details needed to find the functionality in the code base. 
![image](https://user-images.githubusercontent.com/58790905/143308549-f09187b9-9ee1-487f-987e-43668bf24c44.png)

**Comment by LafLaurine** [comment]
Tested on development branch. The described feature, i.e "Right clicking on a Script Canvas node will additionally have the option of something like "open source file"." is still not implemented, but the tooltips are still very useful
